ColorSwitcher/Model/GIFManager.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  GIFManager.swift
+//  ColorSwitcher
+//
+//  Created by Matt M Smith on 9/28/20.
+//
+
+import Foundation
+
+protocol GIFManagerDelegate {
+    func getGIFModel(title: String, urlString: String)
+}
+
+struct GIFManager {
+    var delegate: GIFManagerDelegate?
+    
+    private let apiKey = "0OZKMCWSOD4M"
+    private let baseURL = "https://api.tenor.com/v1/search?"
+    
+    func fetchGIF(from query: String) {
+        let formattedQuery = query.replacingOccurrences(of: " ", with: "+")
+        let urlString = "\(baseURL)?api_key=\(apiKey)&q=\(formattedQuery)&contentfilter=high&media_filter=minimal&limit=1"
+        
+        if let url = URL(string: urlString) {
+            let session = URLSession(configuration: .default)
+            let task = session.dataTask(with: url) { (data, _, error) in
+                if let error = error {
+                    print(error)
+                }
+                if let data = data {
+                    let model = self.parse(jsonData: data)
+
+                    self.delegate?.getGIFModel(title: model.title, urlString: model.gifURLString)
+                }
+            }
+            task.resume()
+        }
+    }
+    
+    private func parse(jsonData data: Data) -> GIFModel {
+        let decoder = JSONDecoder()
+        var title = ""
+        var gifURLString = ""
+        
+        do {
+            let decodedData = try decoder.decode(GIFData.self, from: data)
+            title = decodedData.results[0].title
+            gifURLString = decodedData.results[0].media[0].gif.url
+        } catch {
+            title = "Sorry, but we are having technical Difficulties"
+            gifURLString = Bundle.main.paths(forResourcesOfType: "gif", inDirectory: "GIF")[0]
```

Comment:
Except here where it isn't easy to figure out this is `error.gif`. Does `Bundle.main.path(forResource: "error", ofType: "gif", inDirectory: "GIF")` work?